Sources/NIOKit/Future+Optional.swift

```diff
@@ -0,0 +1,75 @@
+import NIO
+
+extension EventLoopFuture {
+    
+    /// Calls a closure on an optional value in an `EventLoopFuture` if it exists.
+    ///
+    ///     let optional = eventLoop.future(Optional<Int>.some(42))
+    ///     let some = optional.flatMapOptionalThrowing { int -> Float in
+    ///         return int * 3.14
+    ///     }
+    ///     // some: EventLoopFuture(Optional(131.88))
+    ///
+    /// - parameters:
+    ///   - closure: The closure to call on the unwrapped optional value.
+    ///   - unwrapped: The optional's value, unwrapped.
+    /// - returns: The result of the closure if the optional was unwrapped, or nil if it wasn't, wrapped in an `EventLoopFuture`.
+    public func flatMapOptionalThrowing<Wrapped, Result>(_ closure: @escaping (_ unwrapped: Wrapped)throws -> Result?) -> EventLoopFuture<Result?>
```

Comment:
I think these methods should be named the underlying NIO call prefixed with `optional`. That would allow them to be next to each other in code completion and make it clear what to name new methods, too.
So this one should be:
```swift
func optionalFlatMapThrowing
```
Since it calls `flatMapThrowing` internally.